tools/build_api.py

```diff
@@ -1075,20 +1075,7 @@ def get_unique_supported_toolchains(release_targets=None):
                       If release_targets is not specified, then it queries all
                       known targets
     """
-    unique_supported_toolchains = []
-
-    if not release_targets:
-        for target in TARGET_NAMES:
-            for toolchain in TARGET_MAP[target].supported_toolchains:
-                if toolchain not in unique_supported_toolchains:
-                    unique_supported_toolchains.append(toolchain)
-    else:
-        for target in release_targets:
-            for toolchain in target[1]:
-                if toolchain not in unique_supported_toolchains:
-                    unique_supported_toolchains.append(toolchain)
-
-    return unique_supported_toolchains
+    return ["ARM", "uARM", "GCC_ARM", "IAR"]
```

Comment:
To avoid creating a disconnect between the toolchain implementation, can we reference the keys present here instead: https://github.com/ARMmbed/mbed-os/blob/1eab0723bae936988c733fa401c02425e31f3aaa/tools/toolchains/__init__.py#L1112

I realize ARM6 is in that list though. So that would mean just removing it from the list here by name, or adding a "supported" variable (or something similar) to the toolchain class and then filter on that here.